release.sh

```diff
@@ -0,0 +1,289 @@
+    commit_and_push "Release engine $NEW_ENGINE_VERSION" "$BRANCH"
+
+
+    cat <<EOF
+
+[RELEASE] YOUR TURN:
+    0. Double-check my work!
+    1. Make a PR with these changes (CI won't pass)
+    2. Get it approved, but don't merge
+    3. Run \`$0\` again
+EOF
+}
+
+
+stage2() {
+    echo "[RELEASE STAGE 2] 1. Releasing repos"
+    for repo in $REPOS; do
+        cd $repo
+
+        echo "[RELEASE STAGE 2 $repo] 1. Updating deps to fetch freshly released Engine packages"
+        mix deps.update --all
+
+        echo "[RELEASE STAGE 2 $repo] 2. RELEASING"
+        mix hex.publish
+
+        cd ..
+    done
+
+
+    echo "[RELEASE STAGE 2] 2. Committing and pushing changes to the lockfiles"
+    commit_and_push "Update lockfiles"
+
+
+    cat <<EOF
+
+[RELEASE] YOUR TURN:
+    0. Double-check my work!
+    1. Wait for CI to pass (hopefully)
+    2. Merge the PR
+    3. Run \`$0\` again
+EOF
+}
+
+
+stage3() {
+    echo "[RELEASE STAGE 3] 1. Pulling latest changes from 'master'"
+    git checkout master
+    git pull
+
+
+    echo "[RELEASE STAGE 3] 2. Tagging repos"
+    tags=""
+    for repo in $REPOS; do
+        tag="$repo-v$(read_version $repo)"
+        echo "[RELEASE STAGE 3 $repo] Adding tag $tag"
+        git tag $tag
+        tags+=" $tag"
+    done
+
+
+    echo "[RELEASE STAGE 3] 3. Pushing the tags"
+    confirm_execute git push $(git remote) $tags
+
+
+    BRANCH="post-release"
+    echo "[RELEASE STAGE 3] 4. Checking out a new branch '$BRANCH'"
+    create_branch "$BRANCH"
+
+
+    echo "[RELEASE STAGE 3] 5. Updating repos"
+    for repo in $REPOS; do
+        old_version=$(read_version $repo)
+        new_version=$(next_version $old_version)
+        echo "[RELEASE STAGE 3 $repo] 1. Changing version from $old_version to $new_version"
+        sed -i '' -E "s/@version \"$old_version\"/@version \"$new_version\"/" $repo/mix.exs
+
+        echo "[RELEASE STAGE 3 $repo] 2. Adding entry to changelog"
+        printf "# Changelog\n\n## $new_version\n" > $repo/CHANGELOG-GEN.md
+        tail -n+2 $repo/CHANGELOG.md >> $repo/CHANGELOG-GEN.md
+        mv $repo/CHANGELOG-GEN.md $repo/CHANGELOG.md
+
+        echo "[RELEASE STAGE 3 $repo] 3. Changing Engine deps from hex packages to paths"
+        if [[ "$repo" != "engine" ]]; then
+            sed -i '' -E "s/$(dep engine ".*")/$(dep engine path)/" $repo/mix.exs
+
+            if [[ "$repo" != "webrtc" ]]; then
+                sed -i '' -E "s/$(dep webrtc ".*")/$(dep webrtc path)/" $repo/mix.exs
+            fi
+        fi
+    done
+
+
+    echo "[RELEASE STAGE 3] 6. Committing and pushing changes"
+    commit_and_push "Post-release changes" "$BRANCH"
+
+
+    cat <<EOF
+
+[RELEASE] YOUR TURN:
+    0. Double-check my work!
+    1. Make a PR with these changes
+    2. Merge it after getting approvals (no rush)
+EOF
+}
+
+
+case "$1" in
+    -h|--help)
+        usage && exit 0 ;;
+esac
+
+STAGE=$(( $(get_stage) + 1 ))
+echo "[RELEASE] Executing stage $STAGE"
```

Comment:
Shouldn't we check before running a script if the user has some uncommitted changes? Because this script could accidentally commit them.